proposals/experimental.md

```diff
@@ -34,100 +36,113 @@ package kotlin
 // is required to opt-in to that marker. Here, we choose a propagating opt-in,
-// because experimental API is used in the function signature (parameter foo's type).
+// because the API is used in the function signature (parameter foo's type).
 // (But this is not enforced by the compiler in any way, we could've chosen
 // a non-propagating opt-in as well.)
 @ShinyNewAPI
 fun useFooAndBar(foo: Foo) {
     foo.bar()
 }
 
-// Function doSomething uses experimental API with marker ShinyNewAPI and is also
-// required to opt-in. Here, we choose a non-propagating opt-in, because experimental
+// Function doSomething uses API marked with ShinyNewAPI and is also
+// required to opt-in. Here, we choose a non-propagating opt-in, because the
 // API is used in the function body and it should not concern our clients
-@UseExperimental(ShinyNewAPI::class)
+@OptIn(ShinyNewAPI::class)
 fun doSomething() {
     val foo = Foo()
     foo.bar()
 }
 ```
 
-Note that by using the experimental API and using the propagating opt-in, `useFooAndBar` effectively becomes experimental API itself (with the same marker). In theory, we could distinguish initial introduction of the experimental API and its propagating usages, but it would complicate the proposal a bit and there doesn't seem to be much value in doing that.
+Note that by opting into the API with the propagating opt-in, `useFooAndBar` effectively requires opt-in itself (with the same marker). In theory, we could distinguish initial introduction of the API and its propagating usages, but it would complicate the proposal a bit and there doesn't seem to be much value in doing that.
 
 Both opt-in mechanisms allow to use the API for the selected markers anywhere in the parse tree lexically under the annotated element.
 
-Using `UseExperimental` with annotations that are *not* experimental API markers has no effect and yields a compilation warning. (Note that this must not be an error because user code should not break once an annotation is no longer an experimental API marker.) Using `UseExperimental` with no arguments has no effect and yields a warning as well.
+Using `OptIn` with annotations that are *not* opt-in requirement markers has no effect and yields a compilation warning. (Note that this must not be an error because user code should not break once an annotation is no longer an opt-in requirement marker.) Using `OptIn` with no arguments has no effect and yields a warning as well.
+
+## Opt-in requirement message
+
+When using an API that requires opt-in without the said opt-in, the compiler reports a warning or an error depending on the specified level. It's possible to specify a custom message in `RequiresOptIn`, that will be reported by the compiler. If no message is given (i.e. if `message` is empty), the compiler will report that the API is *experimental*:
+
+```
+test.kt:22:9: error: this declaration is experimental and its usage must be marked with '@ShinyNewAPI' or '@OptIn(ShinyNewAPI::class)'
```

Comment:
Why does the default message mention experimentality?
E.g. why not
>est.kt:22:9: error: this declaration required opt-in and its usage must be marked with '@ShinyNewAPI' or '@ OptIn(ShinyNewAPI::class)'

Reply:
Perhaps, to make the migration from `@Experimental` more smooth. 
However we need to think about the message template, so that one who needs a custom message would not have to spell `marked with @Annotation or @OptIn(Annotation::class)` every time.
Let's discuss it later in the corresponding issue, before we start implementing the message support.

Reply:
@qwwdfsad I thought that experimental API is still the main use case and "is experimental" by default is a bit clearer for the user than "requires opt-in". But I'm not very sure about this and can be convinced otherwise.

@ilya-g As far as I remember, we discussed this and explicitly decided to fully replace the message (unlike in Deprecated). See the internal discussion [doc](https://jetbrains.quip.com/Ga4lA7rtaF5Z/DM-Apr-18-2019-Completion-customization-for-experimental-API#XHSACAiFJKj):
> * Add ability to customize diagnostic message in experimental annotations (like in `@Deprecated`). That message will fully replace default message

But if you have a proposal for a message template, let's discuss it.